deepface/models/facial_recognition/Facenet.py

```diff
@@ -67,6 +69,26 @@ def __init__(self):
         self.output_shape = 512
 
 
+class FaceNet512dONNXClient(FacialRecognition):
+    """
+    FaceNet-1512d model class
+    """
+
+    def __init__(self):
+        self.model = load_facenet512d_onnx_model()
+        self.model_name = "FaceNet-512d-onnx"
+        self.input_shape = (160, 160)
+        self.output_shape = 512
+
+    def forward(self, img: np.ndarray) -> List[float]:
+        input_name = self.model.get_inputs()[0].name
+        output_name = self.model.get_outputs()[0].name
+        result = self.model.run([output_name], {input_name: img})
+        if isinstance(result[0][0], np.ndarray):
+            return result[0][0].tolist()
```

Comment:
result[0][0] should always be numpy array, you can get rid of if block and return result[0][0].tolist() if i am not wrong